tools/src/main/scala/scala/scalanative/codegen/Generate.scala

```diff
@@ -155,19 +157,33 @@ object Generate {
           Inst.Let(rt.name, Op.Module(Runtime.name), unwind),
           Inst.Let(arr.name,
                    Op.Call(RuntimeInitSig, RuntimeInit, Seq(rt, argc, argv)),
-                   unwind),
-          Inst.Let(module.name, Op.Module(entry.top), unwind),
-          Inst.Let(Op.Call(entryMainTy, entryMain, Seq(module, arr)), unwind),
-          Inst.Let(Op.Call(RuntimeLoopSig, RuntimeLoop, Seq(module)), unwind),
-          Inst.Ret(Val.Int(0)),
-          Inst.Label(handler, Seq(exc)),
-          Inst.Let(Op.Call(PrintStackTraceSig, PrintStackTrace, Seq(exc)),
-                   Next.None),
-          Inst.Ret(Val.Int(1))
+                   unwind)
         )
+          ++ clinitCalls // call the class initializers before calling main
```

Comment:
Wouldn't it be better to insert this between the `Let`s for `rt` and `arr`? That would avoid the latter two's lifetimes to extend over the calls to the static initializers.